### PR TITLE
Don't crash Use ExpressionBody on local functions in top level statements

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes https://github.com/dotnet/roslyn/issues/57570

### CODE REVIEW

File: src/Analyzers/CSharp/Analyzers/UseExpressionBody/Helpers/UseExpressionBodyForLocalFunctionHelper.cs [hunk 1]
```diff
@@ -45,6 +48,14 @@ protected override LocalFunctionStatementSyntax WithExpressionBody(LocalFunction
         protected override LocalFunctionStatementSyntax WithBody(LocalFunctionStatementSyntax statement, BlockSyntax body)
             => statement.WithBody(body);
 
+        protected override LocalFunctionStatementSyntax GetDeclaration(Location declarationLocation, CancellationToken cancellationToken)
+            => declarationLocation.FindNode(cancellationToken) switch
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Analyzers/UseExpressionBody/Helpers/UseExpressionBodyForLocalFunctionHelper.cs:52]: getInnermostNode: true

File: src/Analyzers/CSharp/CodeFixes/UseExpressionBody/UseExpressionBodyCodeFixProvider.cs [hunk 2]
```diff
@@ -90,7 +89,7 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             var declarationLocation = diagnostic.AdditionalLocations[0];
             var helper = _helpers.Single(h => h.DiagnosticId == diagnostic.Id);
-            var declaration = declarationLocation.FindNode(cancellationToken);
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/UseExpressionBody/UseExpressionBodyCodeFixProvider.cs:93]: Just use getInnermostNode here.

Comment (davidwengier) [reply src/Analyzers/CSharp/CodeFixes/UseExpressionBody/UseExpressionBodyCodeFixProvider.cs:93]: Well thats a lot simpler, thanks!

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**